vcd/provider.go

```diff
@@ -108,6 +109,33 @@ func Provider() terraform.ResourceProvider {
 	}
 }
 
+// This is a global MutexKV for all resources
+var vcdMutexKV = mutexkv.NewMutexKV()
+
+func lockVapp(d *schema.ResourceData) {
+	vcdMutexKV.Lock(d.Get("vdc").(string) + d.Get("name").(string))
```

Comment:
I think we should add TRACE/DEBUG level logging to each of these function with as much details as we can access (maybe a caller function or a resource `name` which are mandatory everywhere anyway?

This should be valuable if we hit any bad situations

Reply:
+1

Reply:
Added